Core/Src/main.c

```diff
@@ -582,12 +582,40 @@ void HAL_DAC_ConvCpltCallbackCh1(DAC_HandleTypeDef *hdac) {
 	HAL_DAC_Stop_DMA(&hdac1, DAC_CHANNEL_1);
 	if (toneIndex >= toneSequenceSize) {
 		playingSequence = 0;
+		toneIndex = 0;
+		HAL_DFSDM_FilterRegularStart_DMA(&hdfsdm1_filter0, currentTone, TONE_LEN);
 		return;
 	}
 	toneIndex++;
 	loadToneFromFlash(toneSequence[toneIndex]);
 	HAL_DAC_Start_DMA(&hdac1, DAC_CHANNEL_1, currentTone, TONE_LEN, DAC_ALIGN_8B_R);
 }
+
+// callback for timer
+void HAL_DFSDM_FilterRegConvCpltCallback (DFSDM_Filter_HandleTypeDef * hdfsdm_filter)
+{
+	// check to make sure correct callback
+	if(hdfsdm_filter == &hdfsdm1_filter0)
+	{
+		HAL_DFSDM_FilterRegularStop_DMA(hdfsdm_filter);
+
+		// scale 24 bits to 8 bit channel
+		for(size_t i = 0; i < TONE_LEN; i++)
+		{
+			currentTone[i] = (currentTone[i] >> 8)/65536 + 128;
+		}
+
+		//write microphone to flash
+		if (BSP_QSPI_Write(currentTone, sizeof(currentTone)*NUM_TONES + toneIndex * TONE_LEN, sizeof(currentTone)) != QSPI_OK) {

```

Comment:
this is a major nitpick (really doesn't matter) but the curly braces are inconsistent -- I personally am a fan of not having them on a new line (for literally exclusively aesthetic purposes) but I understand why people like new line braces. Either way, it would be good to decide on something and stick to it (sorry i know this isnt important im just trying to be thorough)

Reply:
Also this is designed to place the user's recordings after the 4 prerecorded notes right? I *think* TONE_LEN can be replaced with sizeof(currentTone) for consistency's sake, but otherwise this logic looks correct to me

Reply:
Ok no worries, I will fix the curly braces. Correct, I will change this as well